libnavui-shield/src/main/java/com/mapbox/navigation/ui/shield/internal/loader/ResourceDownloader.kt

```diff
@@ -0,0 +1,43 @@
+package com.mapbox.navigation.ui.shield.internal.loader
+
+import com.mapbox.bindgen.Expected
+import com.mapbox.bindgen.ExpectedFactory
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+
+private typealias LoaderCallback<R> = (Expected<String, R>) -> Unit
+
+internal abstract class ResourceDownloader<I, R> : ResourceLoader<I, R> {
```

Comment:
Why don't you use `TileStore` cache? Aren't your needs met by `TileStore`?

Reply:
`RoadShieldDownloader` implicitly uses `TIleStore` via the utils `ResourceLoader` facade.

Reply:
Yes, but isn't the cache `TileStore` provides enough? I can assume it's not because it was not enough for voice instructions. Just want to make sure that writing our own cache above `TileStore` is a conscious choice.

Reply:
> Just want to make sure that writing our own cache above TileStore is a conscious choice.

What's that supposed to mean? First, of course, it is a conscious choice! Second, loading voice instructions don't require parsing large JSON strings with ~400 entries. Third, shields require loading two resources: a) a sprites list; b) images. Finally, the in-memory cache is necessary to avoid redundant I/O requests and computationally heavy result processing.

Reply:
It means that we are aware that `TileStore` provides us with a caching mechanism but we write our own because it doesn't meet our needs.